torchgeo/datamodules/digital_typhoon.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Digital Typhoon Data Module."""
+
+from typing import Any
+
+from ..datasets import DigitalTyphoonAnalysis
+from ..datasets.digital_typhoon import _SampleSequenceDict
+from .geo import NonGeoDataModule
+from .utils import group_shuffle_split
+
+
+class DigitalTyphoonAnalysisDataModule(NonGeoDataModule):
+    """Digital Typhoon Analysis Data Module."""
+
+    valid_split_types = ["time", "typhoon_id"]
+
+    def __init__(
+        self,
+        split_by: str = "time",
+        batch_size: int = 64,
+        num_workers: int = 0,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new DigitalTyphoonAnalysisDataModule instance.
+
+        Args:
+            split_by: Either 'time' or 'typhoon_id', which decides how to split
+                the dataset for train, val, test
+            batch_size: Size of each mini-batch.
+            num_workers: Number of workers for parallel data loading.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.DigitalTyphoonAnalysis`.
+
+        """
+        super().__init__(DigitalTyphoonAnalysis, batch_size, num_workers, **kwargs)
+
+        assert (
+            split_by in self.valid_split_types
+        ), f"Please choose from {self.valid_split_types}"
+        self.split_by = split_by
+
+    def _split_dataset(
+        self, dataset: DigitalTyphoonAnalysis
+    ) -> tuple[list[_SampleSequenceDict], ...]:
```

Comment:
```suggestion
    ) -> tuple[list[_SampleSequenceDict], list[_SampleSequenceDict]]:
```
Tuple will always be length 2